pkg/controllers/provisioning/provisioner.go

```diff
@@ -371,25 +372,22 @@ func (p *Provisioner) Schedule(ctx context.Context) (scheduler.Results, error) {
 		).Info("found provisionable pod(s)")
 	}
 	// Mark in memory when these pods were marked as schedulable or when we made a decision on the pods
-	p.cluster.MarkPodSchedulingDecisions(results.PodErrors, pendingPods...)
-	p.MarkPodSchedulingDecisionsNodeRegistrationHealthy(ctx, results)
+	p.cluster.MarkPodSchedulingDecisions(ctx, results.PodErrors,
```

Comment:
So, if we get a circular dependency when it comes to handling scheduler results, does it make sense to move these results into a separate package so they can be referenced from multiple packages without creating cross-dependencies? I think that might also clean some of this code up

Reply:
Discussed offline: The move to a different package causes us to have to move a _ton_ of other stuff -- ended up not seeming like it was worth it at the end of the day